app/database/admin/data_operator/utils/index.ts

```diff
@@ -200,3 +205,18 @@ export const getRawRecordPairs = (raws: any[]): RecordPair[] => {
         return {raw, record: undefined};
     });
 };
+
+/**
+ * getUniqueRawsBy: We have to ensure that we are not updating the same record twice in the same operation.
+ * Hence, thought it might not occur, prevention is better than cure.  This function removes duplicates from the 'raws' array.
+ * @param {RawValue[]} raws
+ * @param {string} key
+ */
+export const getUniqueRawsBy = ({raws, key}:{ raws: RawValue[], key: string}) => {
+    return [...new Map(raws.map((item) => {
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
```

Comment:
Suppose in a clean installation, we have to create some posts in the database and we get a `Raw` post array as such: [ p1, p1, p2, p3].   WDB already registers `p1` id for CREATION but now it gets `p1` again to create.  The same situation will occur if we attempt to UPDATE `p1` followed by another update on `p1`.  In both cases, it will fail the whole operation and `getUniqueRawsBy` exists for the purpose of sanitizing the data we feed to the operator.

Reply:
I am currently working on a PR for refactoring parts of the DataOperator.  I will be tackling this one as well in it.